### DISCUSSION TITLE
Can i install more than one version of mkdocs on a computer?

### DISCUSSION THREAD

**Discussion by syeda-git** [mkdocs/mkdocs#3097] [Q&A]
We are planning to support two versions of mkdocs in our org. So, wondering if i need two computers to maintain two mkdocs versions or is it possible to install two mkdocs versions on a single computer?

Thanks...
Syed

**Comment by pawamoy** [comment]
I suggest reading about Python virtualenvs which will allow you to install different versions of the same package (MkDocs) on a single computer.

Otherwise Docker would allow the same thing.

**Comment by mur4d1n** [comment]
Well, you can use Python tool virtualenv:

```
pip install virtualenv
python3 venv venv
python3 venv venv1
```

These commands create 2 independent from each other and your "main" Python virtual environments folders

(I highly recommend you creating virtual environment in mkdoocs project root directory)

Then you can just switch 'em using

```
source venv/bin/activate

<some actions in venv>

deactivate

source venv1/bin/activate

<some actions in venv1>

deactivate
```

**Comment by Stwissel** [comment]
If you are OK with docker, the easiest way is to use separate docker containers. It allows to share the setup with others too. In my setup I use the following `Dockerfile` that pre-installs all the plugins.

```docker
# Use mkdocs material
FROM squidfunk/mkdocs-material
RUN apk update && apk upgrade && apk add git
RUN pip install mkdocs-awesome-pages-plugin mkdocs-git-revision-date-localized-plugin mike mkdocs-markdownextradata-plugin mkdocs-git-authors-plugin mkdocs-blog-plugin mkdocs-section-index
EXPOSE 8000
```

Build it with `docker build . -t mkdocs-custom:latest`

The I added a shellscript `~/bin/docs` (I'm on Mac, on Windows you would build a `docs.cmd` somewhere on the path.

```bash
#!/bin/bash
# Run MKDocs in current directory
if [ -f "mkdocs.yml" ] && [ -d "docs" ]; then
   echo Found MKDocs settings
else
   echo initializing MKDocs
   docker run --rm -it -v ${PWD}:/docs mkdocs-custom:latest -q new .   
fi
# actual run
docker run --rm -it -p 8000:8000 -v ${PWD}:/docs mkdocs-custom:latest $1 $2 $3
```

Then I can navigate to any documentation directory and start just by typing `docs`

Repeat that for the other versions using a different name and you are all set.
